libbeat/api/npipe/listener_windows.go

```diff
@@ -0,0 +1,77 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build windows
+
+package npipe
+
+import (
+	"context"
+	"net"
+	"os/user"
+	"strings"
+
+	"github.com/microsoft/go-winio"
+	"github.com/pkg/errors"
+)
+
+// NewListener creates a new Listener receiving events over a named pipe.
+func NewListener(name, sd string) (net.Listener, error) {
+	c := &winio.PipeConfig{
+		SecurityDescriptor: sd,
+	}
+
+	l, err := winio.ListenPipe(name, c)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to listen on the named pipe %s", name)
+	}
+
+	return l, nil
+}
+
+// TransformString takes an input type name defined as a URI like `npipe:///hello` and transform it into
+// `\\.\pipe\hello`
+func TransformString(name string) string {
+	if strings.HasPrefix(s, "npipe:///") {
+		path := strings.TrimPrefix(name, "npipe:///")
+		return `\\.\pipe\` + path
+	}
+
+	if strings.HasPrefix(s, `\\.\pipe\`) {
+		return s
+	}
+
+	return name
+}
+
+// DialContext create a Dial to be use with an http.Client to connect to a pipe.
+func DialContext(npipe string) func(context.Context, string, string) (net.Conn, error) {
+	return func(ctx context.Context, _, _ string) (net.Conn, error) {
+		return winio.DialPipeContext(ctx, npipe)
+	}
+}
+
+// DefaultSD returns a default SecurityDescriptor.
+func DefaultSD() (string, error) {
+	u, err := user.Current()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to retrieve the current user")
+	}
+
+	descriptor := "D:P(A;;GA;;;" + u.Uid + ")"
```

Comment:
this gives user read/write access to the pipe.
maybe we can repalce Generic REad write to: FAFR
also we might think about passing a user id from config/env. this will open a pipe only for a user beat is running as. 
consider an agent starting a beat as UID1 and then a special metricbeat to monitor this beat as UID2. this setting will force UID1==UID2.

Reply:
Would it make sense to make the uid configurable?

We also need some comment (with link to windows docs), why the descriptor is chosen as is.

Reply:
Yeah, I believe it make sense to allow it to be configurable and maybe a fallback to the current using when not configured would be the way to go. I am going to make the change @michalpristas / @elastic/solutions-docs 

Thanks for looking at it, the documentation for the security descriptor is not the best.